evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -1657,3 +1657,29 @@ def test_automl_passes_allow_long_running_models(
         estimators.extend(["CatBoost Classifier", "XGBoost Classifier"])
 
     assert "Dropping estimators {}".format(", ".join(sorted(estimators))) in caplog.text
+
+
+@pytest.mark.parametrize("positive_label", [True, False])
+def test_automl_binary_label_encoder(
+    positive_label,
+    fraud_100,
+    AutoMLTestEnv,
+):
+    X, y = fraud_100
+    pipeline_parameters = {"Label Encoder": {"positive_label": positive_label}}
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        pipeline_parameters=pipeline_parameters,
+        max_batches=1,
```

Comment:
I don't this work if max_batches > 1? 

I just got this from running locally:

```
In [9]:         [
   ...:             p["Label Encoder"]["positive_label"]
   ...:             for p in automl.full_rankings["parameters"][1:]
   ...:         ]
Out[9]: [True, True, None, None, None, None, True, True, True, True, True, True, None]
```

I think we gotta thread this through to the automl algo. This is another example of #3150

Reply:
@freddyaboulton I added some fixes to address this change. Will address #3150 in a separate PR, but I figured it would be nice to get this bug fix in now.